homeassistant/components/foscam/camera.py

```diff
@@ -67,6 +68,20 @@ def camera_image(self):
 
         return response
 
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        return SUPPORT_STREAM
+
+    @property
+    def stream_source(self):
+        """Return the stream source."""
+        return 'rtsp://{}:{}@{}:{}/videoMain'.format(
+            self._username,
+            self._password,
+            self._foscam_session.host,
+            self._foscam_session.port)
```

Comment:
Is this the RTSP port or the port used to connect to the Foscam admin?  You may need to use `self._foscam_session. get_port_info()` to determine what the actual RTSP port is (probably returned as `mediaPort`).  Looking at the upstream library: https://github.com/viswa-swami/python-foscam/blob/master/libpyfoscam/foscam.py#L166

Reply:
added `self._media_port` and grab the media port from the camera.